### ISSUE TITLE
[feature wanted] can i use validate_type macro in my project

### ISSUE THREAD

**Issue by honwhy** [twistedfall/opencv-rust#316]
can i use validate_type macro in my project
valid_types!(ValidRectType: u32);

**Comment by twistedfall** [comment]
Well, theoretically there shouldn't be anything preventing you from using any type with `Rect_` I just wanted to be on the safe side. Can you still please tell me why you need it?

**Comment by honwhy** [comment]
as we use coordinate x and y, (0,0) starts from left-top, maybe it is more precisely using usize type.
I am doing some kind of exercise, [code](https://github.com/honwhy/first_rs/blob/43128b8df0a785ff3b76b1b9878d6ac9806b9f7b/thumbor/src/engine/my_opencv.rs?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L69)
follow the example, which using photon-rs to do image crop,  
```rs
impl SpecTransform<&Crop> for Photon {
    fn transform(&mut self, op: &Crop) {
        let img = transform::crop(&mut self.0, op.x1, op.y1, op.x2, op.y2);
        self.0 = img;
    }
}
```
while I try to implement crop with opencv-rs, I have to type casting
```rs
impl SpecTransform<&Crop> for Opencv {
    fn transform(&mut self, op: &Crop) {
        let rect = Rect_::from_points(
            Point_::<i32>::new(op.x1 as i32, op.y1 as i32),
            Point_::<i32>::new(op.x2 as i32, op.y2 as i32),
        );
        let dest = Mat::roi(&self.0, rect).unwrap();
        self.0 = dest;
    }
}
```

**Comment by twistedfall** [comment]
It should be fixed on 0.63.0, Rect_, Point_, Size_, Point3_, VecN no longer impose restrictions on its container types.
